src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.DecisionDagRewriter.cs

```diff
@@ -148,6 +150,19 @@ public override BoundNode VisitCall(BoundCall node)
                     return null;
                 }
 
+                public override BoundNode VisitPropertyAccess(BoundPropertyAccess node)
+                {
+                    bool mightMutate =
+                        !node.PropertySymbol.IsStatic && node.PropertySymbol.ContainingType.IsStructType() && node.PropertySymbol.ContainingType.SpecialType == SpecialType.None;
```

Comment:
Should we refrain from setting this flag in some cases when the property accessors are `readonly`? Note that it's necessary to check the accessors individually, and if it is not easy to figure out which accessors are in use, it may be expedient to be pessimistic and set '_mightAssignSomething = true' if any accessor on the property is non-readonly.

Reply:
The idea is to be conservative here while being simple.  It is OK of there are opportunities to improve in the future.